pycbc/neutron_stars/pg_isso_solver.py

```diff
@@ -0,0 +1,433 @@
+        for g0, c in zip(initial_guess, chi)])
+    # If the inclination is pi/2, just output the ISSO radius at the pole(s)
+    polar = np.isclose(incl, np.pi / 2)
+    if all(polar):
+        return rISSO_at_pole_limit
+
+    # Otherwise, find the ISSO radius for a generic inclination
+    initial_hi = np.maximum(rISCO_limit, rISSO_at_pole_limit)
+    initial_lo = np.minimum(rISCO_limit, rISSO_at_pole_limit)
+    brackets = [
+        (bl, bh) if c == 1 else None
+        for bl, bh, c in zip(initial_lo, initial_hi, chi)]
+    solution = np.array([
+        root_scalar(
+            PG_ISSO_eq, x0=g0, fprime=PG_ISSO_eq_dr, bracket=bracket,
+            fprime2=PG_ISSO_eq_dr2, args=(c, inc), xtol=1e-12).root
+        for g0, bracket, c, inc in zip(initial_hi, brackets, chi, incl)])
+    oob = (solution < 1) | (solution > 9)
+    if any(oob):
+        solution = np.array([
+            root_scalar(
+                PG_ISSO_eq, x0=g0, fprime=PG_ISSO_eq_dr, bracket=bracket,
+                fprime2=PG_ISSO_eq_dr2, args=(c, inc)).root
+            if ob else sol for g0, bracket, c, inc, ob, sol
+            in zip(initial_lo, brackets, chi, incl, oob, solution)
+            ])
+        oob = (solution < 1) | (solution > 9)
+        if any(oob):
+            raise RuntimeError('Unable to obtain some solutions!')
+    return solution
+
+
+def concat_grid(bounds):
+    '''Constructs non-uniform grid given specified bounds'''
+    out = np.concatenate([
+        np.linspace(0, lim[0], lim[1], endpoint=False) if ii == 0
+        else (
+            np.linspace(bounds[ii-1][0], lim[0], lim[1])
+            if ii == len(bounds)
+            else np.linspace(bounds[ii-1][0], lim[0], lim[1], endpoint=False)
+            )
+        for ii, lim in enumerate(bounds)])
+    return out
+
+
+def generate_isso_bivariate_interp():
+    """Constructs a grid in spin magnitude and spin tilt angle then
+    solves for the ISSO radius (in mass units). Uses the resulting
+    grid to create a bivariate spine which is saved to disk.
+    Note: the grid density need not be uniform, and so has been tuned
+    by hand to be more dense in regions of steeper gradient in order to
+    keep the fractional interpolation error over the full span of
+    parameter values to less than 1e-5.
+    """
+    chis = concat_grid((
+        (0.5, 25), (0.75, 20), (0.95, 30), (0.99, 20), (0.995, 50),
+        (0.9975, 50), (1, 101)))
+    incs = concat_grid((
+        (0.125 * np.pi, 50), (0.375 * np.pi, 300), (0.5 * np.pi, 50),
+        (0.9 * np.pi, 50), (np.pi, 26)))
+    chig, incg = np.meshgrid(chis, incs)
+    roots = np.empty_like(chig)
+    for i, (chi, inc) in enumerate(zip(chig, incg)):
+        # more dense at larger chi
+        roots[i] = PG_ISSO_solver(chi, inc)
+    bivar = RectBivariateSpline(incs, chis, roots)
+    with open(os.path.join(NS_DATA_DIRECTORY, 'isso_inc_chi.pkl'), 'wb') as f:
+        pickle.dump(bivar, f)
```

Comment:
This looks like it is trying to write to a file in a directory within the PyCBC install path, which is generally a horrible idea, and more importantly, will not work in general. For example, it will definitely fail in a IGWN Conda install of PyCBC. It is not clear what the solution is here, maybe the function should take the output file path?

Reply:
Also, a file with the name `isso_inc_chi.pkl` is being committed by this PR. Was that file generated by calling this function?